subsys/pm/pm.c

```diff
@@ -107,26 +136,39 @@ static void pm_resume_devices(void)
 #endif  /* !CONFIG_PM_DEVICE_RUNTIME_EXCLUSIVE */
 #endif	/* CONFIG_PM_DEVICE */
 
+/* Retrieve the element index of a power state. */
+uint8_t pm_state_get_index(enum pm_state state, uint8_t substate_id)
```

Comment:
`static uint8_t pm_state_get_index(enum pm_state state, uint8_t substate_id)`

Reply:
Good call.

On a side note: any thoughts on this and the `static const struct pm_state_info pmstates[]` above living in `state.{c,h}` and becoming part of the API? I think there's likely some use cases outside of the notifier stuff for it.

Reply:
yep (look my comment about data duplication). Another thing, this function is a good candidate to be inlined.